v1/kubernetes.go

```diff
@@ -0,0 +1,550 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/brevdev/cloud/internal/errors"
+)
+
+// Cluster represents the complete specification of a Brev Kubernetes cluster.
+type Cluster struct {
+	// The ID assigned by the cloud provider to the cluster.
+	id CloudProviderResourceID
+
+	// The name of the cluster, displayed on clients.
+	name string
+
+	// The unique ID used to associate with this cluster.
+	refID string
```

Comment:
is the implication that this is a Brev ID? or said another way, what is the difference between `id` and `refId`?

Reply:
ID is the ID assigned by the cloud provider to the cluster, whereas refID is the incoming unique ID used to associate with this cluster (set in the settings).